src/main/java/edu/njit/jerse/automation/GitUtils.java

```diff
@@ -0,0 +1,54 @@
+package edu.njit.jerse.automation;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * The {@code GitUtils} class contains static methods for interacting with Git repositories
+ * using the JGit library.
+ */
+public class GitUtils {
+    private static final Logger LOGGER = LogManager.getLogger(GitUtils.class);
+
+    /**
+     * Clones a Git repository from a specified remote URL into a local directory.
+     * The method allows specifying a particular branch to clone. The repository URL can contain
+     * '.git' at the end, but it is not required.
+     *
+     * @param repoUrl   the URL of the remote repository to clone
+     * @param branch    the branch of the repository to clone. If null, the default branch is cloned.
+     * @param directory the directory where the repository will be cloned into. It should not exist or
+     *                  be an empty directory.
+     * @throws GitAPIException if any error occurs during the cloning process
+     */
+    public static void cloneRepository(String repoUrl, String branch, File directory) throws GitAPIException {
```

Comment:
Minor: I would change `String branch` to `@Nullable String branch`.

Reply:
I have made the change and will push it to this PR - can you explain why this will be beneficial?

Reply:
In the future, we may wish to run the Checker Framework on our own code.  This will ease that.